elrond.md

```diff
@@ -209,13 +97,94 @@ The length is the number of bytes the argument represents.
  // ------------------------------------------------------------
     rule valueArg (arg(V, _)) => V
     rule lengthArg(arg(_, L)) => L
+
+endmodule
+```
+
+Auto Allocate Host Modules
+--------------------------
+
+When `AUTO-ALLOCATE` is imported, an new module will be automatically created and registered whenever necessary to resolve an import.
+This makes it possible to implement host modules easily in K.
+Accessing the import will result in an instruction being left on the `instrs` cell that can't be resolved in the regular Wasm semantics.
+Instead, the embedder can add rules for handling the host import.
+
+Currently, only function imports are supported.
+Calling an imported host function will result in `hostCall(MODULE_NAME, FUNCTION_NAME, FUNCTION_TYPE)` being left on the `instrs` cell.
+
+```k
+module AUTO-ALLOCATE
```

Comment:
Maybe it's better if we call this `WASM-AUTO-ALLOCATE`, as a form of namespacing